File: backend/src/main/java/com/jinanmuseum/entity/Exhibition.java
package com.jinanmuseum.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("exhibition")
public class Exhibition {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String title;
    private String kind; // PERMANENT | TEMPORARY
    private String hall;
    private String summary;
    private String content;
    private String coverImage;
    private LocalDate startDate;
    private LocalDate endDate;
    private Boolean published;
    private Integer sortOrder;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/src/api/exhibition.ts
import api from './index';

export interface Exhibition {
  id: number;
  title: string;
  kind: string;
  hall: string;
  summary: string;
  content: string;
  coverImage: string;
  startDate: string;
  endDate: string;
  published: boolean;
  sortOrder: number;
}

export const getExhibitions = (kind?: string) => {
  return api.get<Exhibition[]>('/exhibitions', { params: { kind } });
};

export const getExhibitionById = (id: number) => {
  return api.get<Exhibition>(`/exhibitions/${id}`);
};